TEveEventDisplay/src/TEveMu2eTracker.cc

```diff
@@ -130,7 +113,7 @@ namespace mu2e{
           CLHEP::Hep3Vector diskPos = calo->disk(idisk).geomInfo().origin() - _detSysOrigin;
           diskPos += CLHEP::Hep3Vector(0.0, 10000.0, (-holeDZ+frontPanelHalfThick));
           double k = diskOuterRailOut + diskInnerRingIn;
-          for(int ic=0; ic<30; ic++){
+          for(int ic=0; ic<calo->nCrystal()/45; ic++){
             CLHEP::Hep3Vector crystalposition(diskPos.x(), (diskPos.y() + k), diskPos.z());
```

Comment:
I must be missing something; it doesn't seem as if the loop variable 'ic' is used inside the loop.

Reply:
OK I see, you are decrementing 'k' each iteration.
This loop needs to be refactored to make the intent more clear.  For instance, the Hep3Vector is never used as such, it would be clearer to increment crystalpos[1] directly, without using 'k' (BTW, i,j, and k are usually used to name integers, not floats).  The crystalpos array should be defined outside the loop, then the x and z components don't need to be reset each iteration as they don't change.   Some comments about what is being done and why n/45 is the correct number are also needed.  Making this a 'while' loop, stopping when crystalpos[1] reaches some physical value, would be less obscure than an integer count.  Please contact Sophie for guidance.